host/xtest/pkcs11_1000.c

```diff
@@ -48,3 +49,81 @@ static void xtest_tee_test_1000(ADBG_Case_t *c)
 
 ADBG_CASE_DEFINE(pkcs11, 1000, xtest_tee_test_1000,
 		"Initialize and close Cryptoki library");
+
+static void xtest_tee_test_1001(ADBG_Case_t *c)
+{
+	CK_RV rv = CKR_GENERAL_ERROR;
+	CK_SLOT_ID_PTR slot_ids = NULL;
+	CK_ULONG slot_count = 0;
+	CK_ULONG present_slot_count = 0;
+	CK_INFO lib_info = { };
+	CK_SLOT_INFO slot_info = { };
+	CK_FUNCTION_LIST_PTR ckfunc_list = NULL;
+	size_t i = 0;
+
+	rv = C_Initialize(NULL);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		return;
+
+	rv = C_GetFunctionList(&ckfunc_list);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	if (!ADBG_EXPECT_NOT_NULL(c, ckfunc_list->C_GetInfo) ||
+	    !ADBG_EXPECT_NOT_NULL(c, ckfunc_list->C_GetSlotList) ||
+	    !ADBG_EXPECT_NOT_NULL(c, ckfunc_list->C_GetSlotInfo))
+		goto out;
+
+	rv = C_GetInfo(&lib_info);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	rv = C_GetSlotList(0, NULL, &slot_count);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	if (!ADBG_EXPECT_COMPARE_UNSIGNED(c, slot_count, !=, 0))
+		goto out;
+
+	rv = C_GetSlotList(1, NULL, &present_slot_count);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	if (!ADBG_EXPECT_COMPARE_UNSIGNED(c, slot_count, ==,
+					  present_slot_count))
+		goto out;
+
+	slot_ids = calloc(slot_count, sizeof(CK_SLOT_ID));
+	if (!ADBG_EXPECT_NOT_NULL(c, slot_ids))
+		goto out;
+
+	rv = C_GetSlotList(1, slot_ids, &slot_count);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	for (i = 0; i < slot_count; i++) {
+		CK_SLOT_ID slot = *(slot_ids + i);
```

Comment:
I'm starting to get used to people using `&ptr[idx]` instead `(ptr + idx)` as a way of hiding the pointer arithmetics. This is the actually the opposite using pointer arithmetics to avoid array indexing.

How about writing `slot_idx[i]` instead since we're actually dereferencing too.

Reply:
right :)  sorry.

Reply:
No problem. :-)